### PR TITLE
Implement nanmedian function in keras.ops

### PR DESCRIPTION
**Pull request by shashaka**
Adds keras.ops.nanmedian, which computes the median of the input tensor along a specified axis while ignoring NaN values.
Supported across NumPy, TensorFlow, PyTorch, and JAX backends. Not supported on OpenVINO.

### CODE REVIEW

File: keras/src/backend/numpy/numpy.py [hunk 1]
```diff
@@ -1084,6 +1084,11 @@ def nanmean(x, axis=None, keepdims=False):
     return np.nanmean(x, axis=axis, keepdims=keepdims).astype(dtype)
 
 
+def nanmedian(x, axis=None, keepdims=False):
+    dtype = dtypes.result_type(standardize_dtype(x.dtype), float)
+    return np.nanmedian(x, axis=axis, keepdims=keepdims).astype(dtype)
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/backend/numpy/numpy.py:1089]: No error occurs in the NumPy backend when axis is () or [].

File: keras/src/ops/numpy_test.py [hunk 2]
```diff
@@ -6482,6 +6512,48 @@ def test_nanmean(self):
             np.nanmean(x_3d, axis=(1, 2)),
         )
 
+    def test_nanmedian(self):
+        x = np.array(
+            [[1.0, np.nan, 3.0, 4.0, 5.0], [np.nan, 2.0, 3.0, np.inf, -np.inf]]
+        )
+
+        self.assertAllClose(knp.nanmedian(x), np.nanmedian(x))
+        self.assertAllClose(knp.nanmedian(x, axis=()), np.nanmedian(x, axis=()))
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/ops/numpy_test.py:6590]: No failed.

File: keras/src/backend/torch/numpy.py [hunk 3]
```diff
@@ -1436,6 +1436,15 @@ def nanmean(x, axis=None, keepdims=False):
     return torch.nanmean(cast(x, dtype), dim=axis, keepdim=keepdims)
 
 
+def nanmedian(x, axis=None, keepdims=False):
+    x = convert_to_tensor(x)
+
+    if axis == () or axis == []:
+        return x
+
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/backend/torch/numpy.py:1444]: The value returned when axis=() differs for each method. For nanquantile, infinite values are treated as NaN, whereas nanmedian returns the original values unchanged.

File: keras/src/backend/torch/numpy.py [hunk 4]
```diff
@@ -1436,6 +1436,15 @@ def nanmean(x, axis=None, keepdims=False):
...
+    return nanquantile(x, q=0.5, axis=axis, keepdims=keepdims)
+
```
Comment (hertschuh) [inline keras/src/backend/torch/numpy.py:1446]: Why not `torch.nanmedian`?

https://docs.pytorch.org/docs/stable/generated/torch.nanmedian.html

**Review by hertschuh — COMMENTED** [review]
Thanks for adding this!

Can you also rebase?

File: keras/src/backend/torch/numpy.py (hunk 4 above)
Comment (shashaka) [reply keras/src/backend/torch/numpy.py:1446]: torch.nanmedian is not used here because its behavior differs from NumPy in key cases.
Specifically, for even-sized inputs, PyTorch returns the lower of the two middle values, whereas NumPy returns their average. 

For example:
```
import numpy as np
import torch

x = np.array([1.0, 3.0])

np.nanmedian(x), torch.nanmedian(torch.tensor(x))
# (np.float64(2.0), tensor(1., dtype=torch.float64))
```

**Review by hertschuh — APPROVED** [review]
Thanks!
